capa/render/proto/capa.proto

```diff
@@ -204,93 +203,67 @@ message Metadata {
 message MnemonicFeature {
   string type = 1;
   string mnemonic = 2;
-  string description = 3;
+  optional string description = 3;
 }
 
 message NamespaceFeature {
   string type = 1;
   string namespace = 2;
-  string description = 3;
+  optional string description = 3;
 }
 
 message NumberFeature {
   string type = 1;
-  oneof number {
-    Integer v0 = 2;
-    Number v1 = 3;
-  };
-  string description = 5;
+  Number number = 2;  // this can be positive (range: u64), negative (range: i64), or a double.
+  optional string description = 5;
 }
 
 message OSFeature {
   string type = 1;
   string os = 2;
-  string description = 3;
+  optional string description = 3;
 }
 
 message OffsetFeature {
   string type = 1;
-  Integer offset = 2;
-  string description = 3;
+  Integer offset = 2;  // offset can be negative
+  optional string description = 3;
 }
 
 message OperandNumberFeature {
   string type = 1;
-  Integer index = 2;
-  Integer operand_number = 3;
-  string description = 4;
+  uint32 index = 2;
+  Integer operand_number = 3;  // this can be positive (range: u64), negative (range: i64), or a double.
+  optional string description = 4;
 }
 
 message OperandOffsetFeature {
   string type = 1;
-  Integer index = 2;
+  uint32 index = 2;
   Integer operand_offset = 3;
-  string description = 4;
+  optional string description = 4;
 }
 
 message PropertyFeature {
   string type = 1;
-  string access = 2;
-  string property = 3;
-  string description = 4;
+  string property_ = 2;  // property is a Python top-level decorator name
+  optional string access = 3;
+  optional string description = 4;
 }
 
 message RangeStatement {
-  string description = 1;
-  Integer min = 2;
-  Integer max = 3;
-  oneof child {
-    OSFeature v0 = 4;
-    ArchFeature v1 = 5;
-    FormatFeature v2 = 6;
-    MatchFeature v3 = 7;
-    CharacteristicFeature v4 = 8;
-    ExportFeature v5 = 9;
-    ImportFeature v6 = 10;
-    SectionFeature v7 = 11;
-    FunctionNameFeature v8 = 12;
-    SubstringFeature v9 = 13;
-    RegexFeature v10 = 14;
-    StringFeature v11 = 15;
-    ClassFeature v12 = 16;
-    NamespaceFeature v13 = 17;
-    APIFeature v14 = 18;
-    PropertyFeature v15 = 19;
-    NumberFeature v16 = 20;
-    BytesFeature v17 = 21;
-    OffsetFeature v18 = 22;
-    MnemonicFeature v19 = 23;
-    OperandNumberFeature v20 = 24;
-    OperandOffsetFeature v21 = 25;
-    BasicBlockFeature v22 = 26;
-  };
-  string type = 28;
+  string type = 1;
+  uint64 min = 2;
+  uint64 max = 3;
+  // reusing FeatureNode here to avoid duplication and list all features OSFeature, ArchFeature, ... again.
+  FeatureNode child = 4;
```

Comment:
This differs slightly from the JSON and was the reason I had set `type = ""` before.
I'd still prefer to not redefine all features here though.